test_app/tests/lib/utils/test_encryption.py

```diff
@@ -43,3 +46,56 @@ def test_fernet256_unsupported_algorithm():
     with pytest.raises(ValueError) as e:
         fernet.decrypt_string(encrypted_string)
     assert str(e.value) == "Unsupported algorithm: monkey"
+
+
+# Define the parameterized test function
+@pytest.mark.parametrize(
+    "input_value, expected",
+    [
+        ('asdf1234', (False, None, None)),
+        ('$encrypted', (False, None, None)),
+        ('$encrypted$something', (False, None, None)),
+        ('$encrypted$UTF8$', (False, None, None)),
+        ('$encrypted$UTF8$asdf1234', (False, None, None)),
+        ('$encrypted$asdf1234', (False, None, None)),
+        ('$encrypted$UTF8$AESCBC$', (False, None, None)),
+        ('$encrypted$AESCBC$', (False, None, None)),
+        ('$encrypted$AESCBC$not4characters', (False, None, None)),
+        ('$encrypted$UTF8$AESCBC$junk', (True, 'AESCBC', 'junk')),
+        ('$encrypted$UTF8$AESCBC$junk12==', (True, 'AESCBC', 'junk12==')),
+        ('$encrypted$UTF8$AESCBC$junk123=', (True, 'AESCBC', 'junk123=')),
+        ('$encrypted$UTF8$AESCBC$junk123=asdf', (False, None, None)),
+        ('$encrypted$UTF8$AESCBC$junk123a', (True, 'AESCBC', 'junk123a')),
+        (smart_str(base64.b64encode(b"This is a test")), (False, None, None)),
+        (f'$encrypted$UTF8$AESCBC${smart_str(base64.b64encode(b"This is a test"))}', (True, 'AESCBC', smart_str(base64.b64encode(b"This is a test")))),
+        (f'$encrypted$AESCBC${smart_str(base64.b64encode(b"This is a test"))}', (True, 'AESCBC', smart_str(base64.b64encode(b"This is a test")))),
+        (
+            f'$encrypted$some_other_algo${smart_str(base64.b64encode(b"This is a test"))}',
```

Comment:
This is an awesome set of tests.  Perhaps for the valid data versions, the string could be varied such that it produces base64 with no padding, single =, and double ==?

Reply:
yes, we have tests for those which are `junk`, `junk123=`, and `junk12==`

Reply:
I meant the opposite actually, "This is a test" -> VGhpcyBpcyBhIHRlc3Q= but one could be changed to "This is a test." -> VGhpcyBpcyBhIHRlc3QuCg== and "This is not a test" -> VGhpcyBpcyBub3QgYSB0ZXN0

The difference is just decrypting valid base64 that has all the possible endings.  But no big deal of course, just a small suggestion to round it out.

Reply:
ahh, "junk" is actually a valid base64 ending. we can do base64.b64decode(b'junk')